scenario-compiler/src/main/scala/pl/touk/nussknacker/engine/definition/component/ComponentDefinitionExtractor.scala

```diff
@@ -73,35 +73,60 @@ object ComponentDefinitionExtractor {
         case other => throw new IllegalStateException(s"Not supported Component class: ${other.getClass}")
       }
 
-    val componentId = ComponentId(componentType, componentName)
-
-    def additionalConfigFromProvider(overriddenDesignerWideId: Option[DesignerWideComponentId]) = {
-      val designerWideId = overriddenDesignerWideId.getOrElse(determineDesignerWideId(componentId))
-
+    def additionalConfigFromProvider(designerWideId: DesignerWideComponentId) = {
       additionalConfigsFromProvider
         .get(designerWideId)
         .map(ComponentAdditionalConfigConverter.toComponentConfig)
         .getOrElse(ComponentConfig.zero)
-        .copy(
-          componentId = Some(designerWideId)
-        )
     }
 
-    def withUiDefinitionForNotDisabledComponent[T](
+    def configFor(defaultConfig: ComponentConfig, withConfigFromProvider: Boolean) = {
+      val componentId                             = ComponentId(componentType, componentName)
+      val configFromAdditional                    = additionalConfigs.getConfig(componentId)
+      val combinedConfigWithoutConfigFromProvider = configFromAdditional |+| configFromDefinition |+| defaultConfig
+      val designerWideId =
+        combinedConfigWithoutConfigFromProvider.componentId.getOrElse(determineDesignerWideId(componentId))
+
+      val finalConfig = if (withConfigFromProvider) {
+        additionalConfigFromProvider(designerWideId) |+| combinedConfigWithoutConfigFromProvider
+      } else {
+        combinedConfigWithoutConfigFromProvider
+      }
+
+      finalConfig.copy(
+        componentId = Some(designerWideId)
+      )
+    }
+
+    def withUiDefinitionForNotDisabledComponent[T <: ComponentDefinitionWithImplementation](
         returnType: Option[TypingResult]
-    )(f: (ComponentUiDefinition, Map[ParameterName, ParameterConfig]) => T): Option[T] = {
+    )(toComponentDefinition: ResolvedComponentConfig => T): Option[T] = {
       val defaultConfig =
         DefaultComponentConfigDeterminer.forNotBuiltInComponentType(
           componentType,
           returnType.isDefined,
           customCanBeEnding
         )
-      val configFromAdditional                    = additionalConfigs.getConfig(componentId)
-      val combinedConfigWithoutConfigFromProvider = configFromAdditional |+| configFromDefinition |+| defaultConfig
-      val designerWideId                          = combinedConfigWithoutConfigFromProvider.componentId
-      val finalCombinedConfig = additionalConfigFromProvider(designerWideId) |+| combinedConfigWithoutConfigFromProvider
 
-      filterOutDisabledAndComputeFinalUiDefinition(finalCombinedConfig, additionalConfigs.groupName).map(f.tupled)
+      val finalEnrichedConfig = configFor(defaultConfig, withConfigFromProvider = true)
+      val finalRawConfig      = configFor(defaultConfig, withConfigFromProvider = false)
+
+      for {
```

Comment:
I think that sth is not ok in this for comprehension. You would return Some if component is not disabled in raw configuration and in enriched configuration? Why do you think that this conjunction is ok? Why not OR or maybe XOR? :) I think that we should build a different set of component definitions at all.

Reply:
I moved components extraction to a higher level, please check `pl.touk.nussknacker.engine.definition.component.Components`